### PR TITLE
Pretty format kotlin independent from git

### PR DESCRIPTION
**Pull request by macisamuele**
The goal of this PR is to remove the direct `git` dependency that is present into `pretty-format-kotlin`.

The git dependency is needed because while running `ktlint` in autofix mode (so with `--format` argument), the tool does not provide any information around the modified files or un-fixable files.

To workaround this limitation the implementation was "manually" comparing the diff pre and post ktlint execution.
This approach made the hook incompatible with parallel run (as we would check the diff of the whole repo).

This PR tries to use a different approach:
* we run ktlint in check mode such that we can have information about what files are not consistent with the kotlin formatting
* run ktlint with `--format` flag, if autofix is enabled, only on these files

The execution time of the hook will be:
 * the same if no files needs to be fixed
 * a bit longer if files needs to be formatted
    NOTE: The time penalty should be comparable with the time saving of not interacting with git, especially for bigger projects. Moreover it would, not in this PR, allow us to run the hook in parallel which should reduce the overall perceived execution time.

---

An additional change introduced by this PR is the capability of defining the Google Java Formatter or KTlint version to be used via pre-commit hooks arguments. In this way we can test new versions of the external tools before bumping this library and provides freedom to the user of this library to use a different version if they like

Example of usage
```yaml
- repo: https://github.com/macisamuele/language-formatters-pre-commit-hooks
  rev: ...
  hooks:
  - id: pretty-format-java
    args:
    - --google-java-formatter-version
    - '1.5'
  - id: pretty-format-kotlin
    args:
    - --ktlint-version
    - 0.35.0
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,85 +1,115 @@
 Changelog
 =========
 
+unreleased (YYYY-MM-DD)
```
Comment (cortinico) [inline CHANGELOG.md:4]: Should this file be updated in a separate PR?

File: language_formatters_pre_commit_hooks/pretty_format_java.py [hunk 2]
```diff
@@ -23,7 +23,16 @@ def get_url(_version):
                 version=_version,
             )
 
-    return download_url(get_url(version))
+    url_to_download = get_url(version)
+    try:
+        return download_url(get_url(version), 'ktlint{version}.jar'.format(version=version))
```
Comment (cortinico) [inline language_formatters_pre_commit_hooks/pretty_format_java.py:28]: nit: this file has a mixture of single and double quotes. Is this intended?

**Review by cortinico — APPROVED** [review]
LGTM 👍

File: CHANGELOG.md (hunk 1 above)
Comment (macisamuele) [reply CHANGELOG.md:4]: I can remove the CHANGELOG update from this PR and publish a new one for doing it, but we do have users using the pre-commit-hooks from master and it would be nice if we do communicate what changes are they using ;)

File: language_formatters_pre_commit_hooks/pretty_format_java.py (hunk 2 above)
Comment (macisamuele) [reply language_formatters_pre_commit_hooks/pretty_format_java.py:28]: Not really intended, sorry for that.
Going to fix it for now even if I'm planning to move the codebase to use black (so we won't discuss about them anymore)

File: CHANGELOG.md (hunk 1 above)
Comment (macisamuele) [reply CHANGELOG.md]: Moved in #31
